builder/dockerignore/dockerignore.go

```diff
@@ -21,7 +21,8 @@ func ReadAll(reader io.ReadCloser) ([]string, error) {
 
 	for scanner.Scan() {
 		pattern := strings.TrimSpace(scanner.Text())
-		if pattern == "" {
+		// Blank lines or lines starting with # are ignored.
+		if pattern == "" || strings.HasPrefix(pattern, "#") {
```

Comment:
I'm ok with either. I always thought it was odd that Dockerfiles didn't support spaces before comments.